### PR TITLE
Clevo/nv41 structure

### PR DESCRIPTION
**Pull request by Stojek139808**

### CODE REVIEW

File: src/mainboard/clevo/adl-p/variants/nv40pz/hda_verb.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	AZALIA_PIN_CFG(2, 0x04, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x06, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x08, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0a, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0b, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0c, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0d, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0e, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0f, 0x18560010),
+
```
Comment (miczyg1) [inline src/mainboard/clevo/adl-p/variants/nv40pz/hda_verb.c:36]:
```suggestion
```

**Comment by macpijan** [comment]
@miczyg1 You fixed one sy yourself and there are no more comments. The review is still in progress?

**Comment by miczyg1** [comment]
@macpijan yes in progress, I still need to look at service manual.

File: src/mainboard/clevo/adl-p/ramstage.c [hunk 2]
```diff
@@ -19,8 +29,7 @@ void mainboard_silicon_init_params(FSP_S_CONFIG *params)
 	params->PchSerialIoI2cSdaPinMux[1] = 0x1947c606; // GPP_H6
 	params->PchSerialIoI2cSclPinMux[1] = 0x1947a607; // GPP_H7
 
-	params->SataPortDevSlpPinMux[0] = 0x59673e0c; // GPP_H12
-	params->SataPortDevSlpPinMux[1] = 0x5967400d; // GPP_H13
+	params->SataPortDevSlpPinMux[1] = 0x59673e0c; // GPP_H12
```
Comment (miczyg1) [inline src/mainboard/clevo/adl-p/ramstage.c:32]: We have to revert this change as it will not work anyways. The DEVSLP pins must match SATA ports 1 to 1. Would also add a comment that design is broken.

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 3]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPO(GPP_B7, 0, DEEP),
+
+	/* GPP_B8 - GPIO */
+	PAD_CFG_GPO(GPP_B8, 0, DEEP),
+
+	/* GPP_B9 - GPIO */
+	PAD_CFG_GPO(GPP_B9, 0, DEEP),
+
+	/* GPP_B10 - GPIO */
+	PAD_CFG_GPO(GPP_B10, 0, DEEP),
```
Comment (miczyg1) [inline src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c:44]: These do not seem to be present on schematics, shouldn't they be NC instead?

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 4]
```diff
@@ -0,0 +1,881 @@
...
+
+	/* GPP_B6 - I2C2_SCL */
+	PAD_CFG_NF(GPP_B6, NONE, DEEP, NF2),
+
+	/* GPP_B7 - GPIO */
+	PAD_CFG_GPO(GPP_B7, 0, DEEP),
+
+	/* GPP_B8 - GPIO */
+	PAD_CFG_GPO(GPP_B8, 0, DEEP),
+
```
Comment (miczyg1) [inline src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c:39]: test points, should be NC?

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 5]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_NF(GPP_B14, NONE, DEEP, NF1),
+
+	/* GPP_B15 - GPIO */
+	PAD_CFG_GPO(GPP_B15, 0, DEEP),
+
+	/* GPP_B16 - GPIO */
+	PAD_CFG_GPO(GPP_B16, 0, DEEP),
+
+	/* GPP_B17 - GPIO */
+	PAD_CFG_GPO(GPP_B17, 0, DEEP),
```
Comment (miczyg1) [inline src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c:65]: Test points, should be NC?

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 6]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPO(GPP_B19, 0, DEEP),
+
+	/* GPP_B20 - GPIO */
+	PAD_CFG_GPO(GPP_B20, 0, DEEP),
+
+	/* GPP_B21 - GPIO */
+	PAD_CFG_GPO(GPP_B21, 0, DEEP),
+
+	/* GPP_B22 - GPIO */
+	PAD_CFG_GPO(GPP_B22, 0, DEEP),
```
Comment (miczyg1) [inline src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c:80]: These do not seem to be present on schematics, shouldn't they be NC instead?

File: src/mainboard/clevo/adl-p/variants/nv40pz/overridetree.cb [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+		end
+		device ref pcie_rp5 on
+			# PCIe RP#5 x1, Clock 2 (WLAN)
+			register "pch_pcie_rp[PCH_RP(5)]" = "{
+				.clk_src = 2,
+				.clk_req = 2,
+				.flags = PCIE_RP_LTR | PCIE_RP_AER,
+			}"
+			chip drivers/wifi/generic
+				register "wake" = "GPE0_DW2_02" # GPD2 shared with LAN
```
Comment (miczyg1) [inline src/mainboard/clevo/adl-p/variants/nv40pz/overridetree.cb:146]:
```suggestion
				register "wake" = "GPE0_PME_B0"
```
WLAN_WAKEUP# signal is not used actually, there is an unpopulated resistor there (marked wit hasterisk). The real wake signal goes to PCIE_WAKE# which corresponds to PME_B0. The same should be done for NS50/70PU

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c (hunk 4 above)
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c (hunk 3 above)
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Set GPP_B10 as NC

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c (hunk 5 above)
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c (hunk 6 above)
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 8]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_NC(GPP_T0, NONE),
+
+	/* GPP_T1 - GPIO */
+	PAD_NC(GPP_T1, NONE),
+
+	/* GPP_T2 - GPIO */
+	PAD_CFG_GPO(GPP_T2, 0, DEEP),
+
+	/* GPP_T3 - GPIO */
+	PAD_CFG_GPO(GPP_T3, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 9]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_NF(GPP_A5, UP_20K, DEEP, NF1),
+
+	/* GPP_A6 - GPIO */
+	PAD_CFG_GPO(GPP_A6, 0, DEEP),
+
+	/* GPP_A7 - GPIO */
+	PAD_CFG_GPO(GPP_A7, 0, DEEP),
+
+	/* GPP_A8 - GPIO */
+	PAD_CFG_GPO(GPP_A8, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 10]
```diff
@@ -0,0 +1,881 @@
...
+
+	/* GPP_A9 - ESPI_CLK */
+	PAD_CFG_NF(GPP_A9, DN_20K, DEEP, NF1),
+
+	/* GPP_A10 - ESPI_RESET# */
+	PAD_CFG_NF(GPP_A10, NONE, DEEP, NF1),
+
+	/* GPP_A11 - GPIO */
+	PAD_CFG_GPO(GPP_A11, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 11]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPI_INT(GPP_A17, NONE, PLTRST, LEVEL),
+
+	/* GPP_A18 - DDSP_HPDB */
+	PAD_CFG_NF(GPP_A18, NONE, DEEP, NF1),
+
+	/* GPP_A19 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_A19, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_A20 - GPIO */
+	PAD_CFG_GPO(GPP_A20, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 12]
```diff
@@ -0,0 +1,881 @@
...
+
+	/* GPP_A21 - GPIO */
+	PAD_CFG_GPO(GPP_A21, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 13]
```diff
@@ -0,0 +1,881 @@
...
+
+	/* GPP_A22 - GPIO */
+	PAD_CFG_GPO(GPP_A22, 1, PLTRST),
+
+	/* GPP_A23 - GPIO */
+	PAD_CFG_GPO(GPP_A23, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 14]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPO(GPP_S4, 0, DEEP),
+
+	/* GPP_S5 - GPIO */
+	PAD_CFG_GPO(GPP_S5, 0, DEEP),
+
+	/* GPP_S6 - GPIO */
+	PAD_CFG_GPO(GPP_S6, 0, DEEP),
+
+	/* GPP_S7 - GPIO */
+	PAD_CFG_GPO(GPP_S7, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 15]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_NF(GPP_H10, NONE, DEEP, NF2),
+
+	/* GPP_H11 - UART0_TXD */
+	PAD_CFG_NF(GPP_H11, NONE, DEEP, NF2),
+
+	/* GPP_H12 - GPIO */
+	PAD_CFG_GPO(GPP_H12, 0, DEEP),
+
+	/* GPP_H13 - GPIO */
+	PAD_CFG_GPO(GPP_H13, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 16]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPO(GPP_H19, 0, DEEP),
+
+	/* GPP_H20 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_H20, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_H21 - GPIO */
+	PAD_CFG_GPO(GPP_H21, 0, DEEP),
+
+	/* GPP_H22 - GPIO */
+	PAD_CFG_GPO(GPP_H22, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 17]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPO(GPP_F9, 0, DEEP),
+
+	/* GPP_F10 - GPIO */
+	PAD_CFG_GPO(GPP_F10, 1, DEEP),
+
+	/* GPP_F11 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_F11, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_F12 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_F12, NONE, PLTRST, OFF, ACPI),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 18]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_NF(GPP_E9, NONE, DEEP, NF1),
+
+	/* GPP_E10 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_E10, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_E11 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_E11, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_E12 - GPIO */
+	PAD_CFG_GPO(GPP_E12, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 19]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPI_TRIG_OWN(GPP_E13, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_E14 - DDSP_HPDA */
+	PAD_CFG_NF(GPP_E14, NONE, DEEP, NF1),
+
+	/* GPP_E15 - GPIO */
+	PAD_CFG_GPO(GPP_E15, 0, DEEP),
+
+	/* GPP_E16 - GPIO */
+	PAD_CFG_GPO(GPP_E16, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 20]
```diff
@@ -0,0 +1,881 @@
...
+
+	/* GPP_E17 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_E17, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_E18 - GPIO */
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 21]
```diff
@@ -0,0 +1,881 @@
...
+	//PAD_NC(GPP_E18, NATIVE),
+
+	/* GPP_E19 - GPIO */
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 22]
```diff
@@ -0,0 +1,881 @@
...
+	//PAD_NC(GPP_E19, NATIVE),
+
+	/* GPP_E20 - GPIO */
+	PAD_CFG_GPO(GPP_E20, 0, DEEP),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/variants/nv40pz/overridetree.cb [hunk 23]
```diff
@@ -0,0 +1,194 @@
...
+				.clk_req = 0,
+				.flags = PCIE_RP_LTR,
+			}"
+			chip soc/intel/common/block/pcie/rtd3
+				register "enable_gpio" = "ACPI_GPIO_OUTPUT_ACTIVE_HIGH(GPP_D14)" # SSD1_PWR_EN
+				register "reset_gpio" = "ACPI_GPIO_OUTPUT_ACTIVE_LOW(GPP_F20)" # M2_SSD1_RST#
+				register "srcclk_pin" = "0" # SSD1_CLKREQ#
+				device generic 0 on end
+			end
+		end
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/overridetree.cb:29]: dGPU not supported at the moment

File: src/mainboard/clevo/adl-p/ramstage.c (hunk 2 above)
Comment (mkopec) [reply src/mainboard/clevo/adl-p/ramstage.c:32]: Added

File: src/mainboard/clevo/adl-p/Kconfig [hunk 24]
```diff
@@ -4,7 +4,7 @@ config BOARD_CLEVO_ADLP_COMMON
 	select DRIVERS_I2C_HID
 	select DRIVERS_INTEL_PMC
 	select DRIVERS_INTEL_USB4_RETIMER
-	select EC_CLEVO_EC
+	select EC_SYSTEM76_EC
```
Comment (macpijan) [inline src/mainboard/clevo/adl-p/Kconfig:7]: Maybe for upstream EC_CLEVO_EC should be selected here by default, and be configurable (also changed in NovaCustom directory)?

**Comment by macpijan** [comment]
@mkopec NovaCustom mainboard entry is not added here as we have added for NS50/70.

File: src/mainboard/clevo/adl-p/variants/nv40pz/hda_verb.c [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+	AZALIA_PIN_CFG(2, 0x04, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x06, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x08, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0a, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0b, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0c, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0d, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0e, 0x18560010),
+	AZALIA_PIN_CFG(2, 0x0f, 0x18560010),
+};
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/hda_verb.c:36]: Copied in https://github.com/Dasharo/coreboot/commit/dda02f99ff30fdf2d2063f0e0eb4e6eacc580b38

**Comment by macpijan** [comment]
@mkopec Config is missing. Both for clevo and NovaCustom.

**Comment by mkopec** [comment]
> @mkopec NovaCustom mainboard entry is not added here as we have added for NS50/70.

Added

> @mkopec Config is missing. Both for clevo and NovaCustom.

Added both

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 26]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPO(GPP_A13, 1, PLTRST),
+
+	/* GPP_A14 - GPIO */
+	PAD_CFG_GPO(GPP_A14, 1, DEEP),
+
+	/* GPP_A15 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_A15, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_A16 - USB_OC3# */
+	PAD_CFG_NF(GPP_A16, NONE, DEEP, NF1),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c:192]: Weird, the schematics clearly label these as OC, but they seem not to be used for anything indeed. Should I just change it to NC?

[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c:192]: Yes

[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: done

File: src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c [hunk 27]
```diff
@@ -0,0 +1,881 @@
...
+	PAD_CFG_GPI_TRIG_OWN(GPP_E6, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_E7 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_E7, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_E8 - GPIO */
+	PAD_CFG_GPI_TRIG_OWN(GPP_E8, NONE, DEEP, OFF, ACPI),
+
+	/* GPP_E9 - USB_OC0# */
+	PAD_CFG_NF(GPP_E9, NONE, DEEP, NF1),
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/clevo/adl-p/variants/nv40pz/gpio.c]: Done

File: src/mainboard/clevo/adl-p/Kconfig (hunk 24 above)
Comment (macpijan) [reply src/mainboard/clevo/adl-p/Kconfig:7]: let's go back to it
